src/main/java/org/folio/circulation/domain/OverdueFineCalculatorService.java

```diff
@@ -57,15 +57,34 @@ private OverdueFineCalculatorService(Clients clients) {
     );
   }
 
-  public CompletableFuture<Result<FeeFineAction>> createOverdueFineIfNecessary(
+  public CompletableFuture<Result<RenewalContext>> createOverdueFineIfNecessary(
     RenewalContext context) {
 
-    Loan loan = context.getLoanBeforeRenewal();
+    final String loggedInUserId = context.getLoggedInUserId();
+    final Loan loanBeforeRenewal = context.getLoanBeforeRenewal();
+
+    return shouldChargeOverdueFineOnRenewal(context)
+      .thenCompose(r -> r.afterWhen(ResultBinding.toFutureResult(),
```

Comment:
Personally, I'd statically import this to reduce the noise

```suggestion
      .thenCompose(r -> r.afterWhen(toFutureResult(),
```